lessons/misc/ml-intro/kmeans.Rmd

```diff
@@ -0,0 +1,117 @@
+---
+title: Unsupervised Learning
```

Comment:
I would recommend putting both `kmeans.Rmd` and `linear_reg.Rmd` files into a single `lessons.Rmd` file. Please see my comment about moving files to `lessons/r/intro-ml` and about the `template/lessons.md` file.